include/cantera/kinetics/Kinetics.h

```diff
@@ -812,6 +849,24 @@ class Kinetics
             "Not implemented for kinetics type '{}'.", kineticsType());
     }
 
+    /**
+     * Calculate derivatives for net rates-of-progress with respect to species
+     * concentration at constant temperature, pressure, and remaining species
+     * concentrations.
+     *
+     * The method returns a matrix with nReactions rows and nTotalSpecies columns.
+     * For a derivative with respect to \f$[n_i]\f$, all other \f$[n_i]\f$ are held
+     * constant.
+     *
+     * @warning  This method is an experimental part of the %Cantera API and
+     *      may be changed or removed without notice.
+     */
+    virtual Eigen::SparseMatrix<double> netRatesOfProgress_ddN()
```

Comment:
@speth I agree that we need a means of clearly distinguishing the two, I used `ddN` because `ddC` was already taken. `ddN` isn't exactly great though as you mentioned.  I think `ddc` is most logical but would probably lead to a lot of misusage with the only difference being a capitalization.  We could make it something like `ddCi` to specify a specific concentration of a specific component.